internal/clients/sources/sources_impl.go

```diff
@@ -2,14 +2,75 @@ package sources
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
 
 	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
 )
 
+var appTypeId string
+var lazyLoad sync.Mutex
+
+type AppType struct {
+	Id          string `json:"id"`
+	Name        string `json:"name"`
+	DisplayName string `json:"display_name"`
+}
+
+type dataElement struct {
+	Data []AppType `json:"data"`
+}
+
+func getAppTypeId() string {
+	lazyLoad.Lock()
+	defer lazyLoad.Unlock()
+	return appTypeId
+}
+
 func getSourcesClient(ctx context.Context) (SourcesIntegration, error) {
-	return NewClientWithResponses(config.Sources.URL)
+	client, err := NewClientWithResponses(config.Sources.URL)
+	if err != nil {
+		return client, err
+	}
+	if err = lazyLoadAppId(ctx, client); err != nil {
+		return client, err
```

Comment:
Isn't better to return `nil, err`? I mean, the caller will unlikely be able to continue.